sdk/servicebus/service-bus/src/core/linkEntity.ts

```diff
@@ -224,35 +267,34 @@ export class LinkEntity {
   }
 
   /**
-   * Closes the Sender|Receiver link and it's underlying session and also removes it from the
-   * internal map.
+   * Closes the internally held rhea link, stops the token renewal timer and sets
+   * the this._link field to undefined.
    *
-   * @param {Sender | Receiver} [link] The Sender or Receiver link that needs to be closed and
-   * removed.
+   * @param mode Indicates the original caller.
+   * - "permanently" closes the link permanently, setting _wasClosedByUser to true which
```

Comment:
From what I have seen, `closeLink("permanently")` is called when `close()` is called on the sender/receiver/mgmtLink which can happen in the below cases:
- User calls close() on the ServiceBusClient or the public facing Sender, Receiver or SessionReceiver
- There is an error on the underlying AMQP receiver link when using sessions, then MessageSession.close() gets called
- Non-retryable error occurs on the underlying AMQP receiver link when not using sessions or when retry attempts are exhausted
- Connection is disconnected and we call close() on all the management links

Are we sure we want to treat all of the above cases as "closed by user"?

Reply:
Very true. I can change it to more neutrally name it as "_isClosed" or something of that nature.

Reply:
Renamed it to _wasClosed.